packages/admin/admin/src/dataGrid/useDataGridRemote.test.ts

```diff
@@ -1,8 +1,202 @@
+        const mockedReplace = mockReplace();
+
+        const { result } = renderHook(() => useDataGridRemote());
+
+        result.current.onPageChange?.(mockedPage, {});
+
+        expect(mockedReplace).toHaveBeenCalledWith({ search: queryString.stringify({ page: mockedPage }) });
+    });
+
+    it("writes pageSize param to URL", () => {
+        const mockedReplace = mockReplace();
+
+        const { result } = renderHook(() => useDataGridRemote());
+
+        result.current.onPageSizeChange?.(mockedPageSize, {});
+
+        expect(mockedReplace).toHaveBeenCalledWith({ search: queryString.stringify({ pageSize: mockedPageSize }) });
+    });
+
+    it("uses queryParamsPrefix when reading params from URL", () => {
+        const prefix = "mocked";
+        (useLocation as jest.Mock).mockReturnValue({
+            search: queryString.stringify({
+                [`${prefix}sort`]: `${mockedSortField}:${mockedSort}`,
+                [`${prefix}filter`]: JSON.stringify(mockedFilterModel),
+                [`${prefix}page`]: mockedPage,
+                [`${prefix}pageSize`]: mockedPageSize,
+            }),
+        });
+
+        const { result } = renderHook(() => useDataGridRemote({ queryParamsPrefix: prefix }));
+
+        expect(result.current.sortModel).toEqual([{ field: mockedSortField, sort: mockedSort }]);
+        expect(result.current.filterModel).toEqual(mockedFilterModel);
+        expect(result.current.page).toBe(mockedPage);
+        expect(result.current.pageSize).toBe(mockedPageSize);
+    });
+
+    it("adds queryParamsPrefix when writing to URL if queryParamsPrefix is set", () => {
+        const prefix = "mocked";
+        const mockedReplace = mockReplace();
+
+        const { result } = renderHook(() => useDataGridRemote({ queryParamsPrefix: prefix }));
+
+        result.current.onSortModelChange?.([{ field: mockedSortField, sort: mockedSort }], {});
+        expect(mockedReplace.mock.calls[0]?.[0]?.search).toContain(`${prefix}sort`);
+
+        result.current.onFilterModelChange?.(mockedFilterModel, {});
+        expect(mockedReplace.mock.calls[1]?.[0]?.search).toContain(`${prefix}filter`);
+
+        result.current.onPageChange?.(mockedPage, {});
+        expect(mockedReplace.mock.calls[2]?.[0]?.search).toContain(`${prefix}page`);
+
+        result.current.onPageSizeChange?.(mockedPageSize, {});
+        expect(mockedReplace.mock.calls[3]?.[0]?.search).toContain(`${prefix}pageSize`);
+
+        expect(mockedReplace).toHaveBeenCalledTimes(4);
+    });
+
+    it("uses initial default pageSize if no pageSize is set in the URL", () => {
+        const mockedInitialPageSize = 37;
+
+        const { result } = renderHook(() => useDataGridRemote({ pageSize: mockedInitialPageSize }));
+
+        expect(result.current.pageSize).toBe(mockedInitialPageSize);
+    });
+
+    it("doesn't use initial default pageSize if a pageSize is set in the URL", () => {
+        const mockedInitialPageSize = 37;
```

Comment:
Nit: I'm not sure if this would qualify as a mock, as it doesn't really mock anything. It's just an initial page size param. 🤔

Reply:
Technically true. I prefixed all variables with `mocked` that I use to store test data.